src/pages/dataTableUtilities/reducer/reducerMethods.js

```diff
@@ -252,3 +262,124 @@ export const sortData = (state, action) => {
   const newData = newSortDataBy(data, newSortBy, columnKeyToDataType[newSortBy], newIsAscending);
   return { ...state, data: newData, sortBy: newSortBy, isAscending: newIsAscending };
 };
+
+/**
+ * Opens a basic modal which does not require any other value
+ * to be set upon opening. i.e. text editors
+ *
+ * @param {Object} state  The current state
+ * @param {Object} action The action to act upon
+ * Action: { type: 'openBasicModal', modalKey }
+ */
+export const openBasicModal = (state, action) => {
+  const { modalKey } = action;
+  return { ...state, modalIsOpen: true, modalKey };
+};
+
+/**
+ * Sets the modal open state to false, closing any
+ * modal that is open.
+ *
+ * @param {Object} state  The current state
+ * Action: { type: 'closeBasicModal' }
+ */
+export const closeBasicModal = state => ({ ...state, modalIsOpen: false, modalKey: '' });
+
+/**
+ * Adds all items from a master list to the pageObject held
+ * in state - either a Requisition or Transaction.
+ * @param {Object} state  The current state
+ * @param {Object} action The action to act upon
+ * Action: { type: 'addMasterListItems', objectType }
+ */
+export const addMasterListItems = (state, action) => {
+  const { pageObject, database, backingData } = state;
+  const { objectType } = action;
+
+  database.write(() => {
+    pageObject.addItemsFromMasterList(database);
+    database.save(objectType, pageObject);
+  });
+
+  const newData = backingData.slice();
+
+  return { ...state, data: newData };
+};
+
+/**
+ * Creates an Item (Either Requisition or Transaction), and appends
+ * this item to the data array for a page.
+ *
+ * @param {Object} state  The current state
+ * @param {Object} action The action to act upon
+ * Action: { type: 'adadItem', item, addedItemType }
+ */
+export const addItem = (state, action) => {
+  const { database, pageObject, data } = state;
+  const { item, addedItemType } = action;
+  let addedItem;
+
+  database.write(() => {
+    if (pageObject.hasItem(item)) return;
+    addedItem = createRecord(database, addedItemType, pageObject, item);
+  });
+
+  if (addedItem) return { ...state, data: [addedItem, ...data], modalIsOpen: false };
+
+  return { ...state, modalIsOpen: false };
+};
+
+/**
+ * Edits the passed pageObject field with the value supplied.
+ * Used for simple value setting i.e. comments.
+ *
+ * @param {Object} state  The current state
+ * @param {Object} action The action to act upon
+ * Action: { type: 'editPageObject', value, field, pageObjectType }
+ */
+export const editPageObject = (state, action) => {
+  const { database, pageObject } = state;
+  const { value, pageObjectType, field } = action;
+
+  if (pageObject[field] !== value) {
+    database.write(() => {
+      pageObject[field] = value;
```

Comment:
Just checked it out. It is possible! Do you think more explicit is better? I'm happy to do that. I figured it would come at some point when the logic between different fields starts to differ

Reply:
I think explicit is better, deleting this method would be mean less invitation to skip being explicit

Reply:
b6575d58661aed26dfe5faca59243fbf1d02adf6
1ad3a1b8d3c3a9e0d03a4cb515cccb1a2ffe342a
3caf7f99901b574ee0e6aec3c89de5ac5169258d
0d04575dbdfd38473fb938025763b700251020e2

added `editTheirRef` and `editComment` reducer methods and actions and removed export of `editPageObject`. I think a `editPageObjectSetter` might be a better idea, rather than check if the `pageObject[field]` is a function, and to call rather than assign, but maybe not. Either way I didn't do the change for `editTotalQuantity`, I think it's fine to figure out a bit later, maybe after a couple of pages to make sure we have a broader scope that: two methods for editting fields/using setters or one and how to run the file system for all these reducers methods/actions (and to consider for not just this, but other hooks that might be used throughout the app, aaaand possibly selectors 🤣 ) in #1092 ?